types/rlp_encoding_test.go

```diff
@@ -162,6 +162,67 @@ func TestRLPMarshall_And_Unmarshall_TypedTransaction(t *testing.T) {
 
 		unmarshalledTx.ComputeHash()
 		assert.Equal(t, originalTx.Type, unmarshalledTx.Type)
+		assert.Equal(t, originalTx.Hash, unmarshalledTx.Hash)
+	}
+}
+
+func TestRLPMarshall_Unmarshall_Missing_Data(t *testing.T) {
+	txTypes := []TxType{
+		StateTx,
+		LegacyTx,
+	}
+
+	arena := fastrlp.DefaultArenaPool.Get()
+	defer fastrlp.DefaultArenaPool.Put(arena)
+
+	for _, txType := range txTypes {
+		testTable := []struct {
+			name          string
+			expectedErr   bool
+			ommitedValues map[string]bool
+			fromAddrSet   bool
+		}{
+			{
+				name:        "Insuficient params",
+				expectedErr: true,
+				ommitedValues: map[string]bool{
+					"Nonce":    true,
+					"GasPrice": true,
+				},
+			},
+			{
+				name:        "Missing From",
+				expectedErr: false,
+				ommitedValues: map[string]bool{
+					"From": true,
+				},
+				fromAddrSet: false,
+			},
+			{
+				name:          "Address set for state tx only",
+				expectedErr:   false,
+				ommitedValues: map[string]bool{},
+				fromAddrSet:   txType == StateTx,
+			},
+		}
```

Comment:
Seems like you could have moved the definition of `testTable` above the for loop?